src/DataGrid.tsx

```diff
@@ -504,8 +548,10 @@ function DataGrid<R, K extends keyof R, SR>({
 
   function selectCell(position: Position, enableEditor = false): void {
     if (!isCellWithinBounds(position)) return;
+    handleCommit2();
```

Comment:
I think it is better to have some control over when the value is committed. This way we can add validation in the future. Using an effect to commit the value is probably not the best way 
https://github.com/adazzle/react-data-grid/blob/c/src/editors/EditorContainer.tsx#L70